producers/http/router.go

```diff
@@ -40,6 +34,18 @@ func newRouter(p *producerImpl) *mux.Router {
 			Name(route.Name).
 			Handler(handler)
 	}
+}
+
+// NewRouter iterates over a slice of Route types and creates them
+// in gorilla/mux.
+func newRouter(p *producerImpl) *mux.Router {
```

Comment:
this looks a bit strange. A name `producerImpl` hints there should be a producer interface? maybe should this function expect an interface rather then implementation?